### PR TITLE
[Enhancement] Support text-based profile analysis(4)

### PR DESCRIPTION
**Pull request by liuyehcf**
## Main work of this pr

This is an continuation work of the previous pr #28813. This pr fixes the NPE when getting the cost extimeation of the expression. The estimation calculation requires the `ConnectContext`, which is missing during the runtime profile prcess.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 14 / 17 (82.35%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|0|2|00.00%|[158, 161]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|[764]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|14|14|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 15 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|[764]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|0|2|00.00%|[968, 970]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|9|11|81.82%|[152, 157]|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 15 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|[764]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|0|2|00.00%|[968, 970]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|9|11|81.82%|[152, 157]|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 17 (58.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[764, 765]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|0|3|00.00%|[969, 971, 972]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|9|11|81.82%|[152, 157]|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 17 (58.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[764, 765]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|0|3|00.00%|[969, 971, 972]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|9|11|81.82%|[152, 157]|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/plan/ExecPlan.java [hunk 1]
```diff
@@ -139,6 +142,27 @@ public OptExpression getOptExpression(int planNodeId) {
         return optExpressions.get(planNodeId);
     }
 
+    public ProfilingExecPlan getProfilingPlan() {
+        if (profilingPlan == null) {
+            synchronized (this) {
+                if (profilingPlan == null) {
+                    boolean isUnset = ConnectContext.get() == null;
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/plan/ExecPlan.java:149]: what scenario doesn't set  connectContext  when execute `explan analyze`?

**Review by Seaven — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 21 (61.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|2|00.00%|[764, 765]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|0|3|00.00%|[969, 971, 972]|
|:large_blue_circle:|com/starrocks/common/util/ProfilingExecPlan.java|3|4|75.00%|[256]|
|:large_blue_circle:|com/starrocks/sql/plan/ExecPlan.java|9|11|81.82%|[152, 157]|
|:large_blue_circle:|com/starrocks/common/util/ProfileManager.java|1|1|100.00%|[]|

**Comment by liuyehcf** [comment]
This pr contains runtime proifle process, which cannot be covered in ut.

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**